burn-import/src/onnx/codegen.rs

```diff
@@ -0,0 +1,578 @@
+            if node.is_stateful {
+                call_nodes.push(Self::node_call_stateful(node));
+            } else {
+                call_nodes.push(Self::node_call_stateless(node, imports));
+            }
+        }
+
+        quote! {
+            pub fn forward(&self, #(#inputs,)*) -> #return_type {
+                #(#call_nodes)*
+                #results
+            }
+        }
+    }
+
+    /// Generates source code for the stateful node calls, i.e. conv, dropout, etc.
+    fn node_call_stateful(node: &Node) -> TokenStream {
+        if !node.is_stateful {
+            panic!("Node must be stateful");
+        }
+
+        let name = Ident::new(&node.name, Span::call_site());
+
+        let mut inputs = vec![];
+
+        for input in node.inputs.iter() {
+            let name = Ident::new(&input.name, Span::call_site());
+            inputs.push(quote! {
+                #name
+            });
+        }
+
+        let mut outputs = vec![];
+
+        for output in node.outputs.iter() {
+            let name = Ident::new(&output.name, Span::call_site());
+            outputs.push(quote! {
+                #name
+            });
+        }
+
+        if outputs.len() == 1 {
+            let output = outputs.pop().unwrap();
+            quote! {
+                let #output = self.#name.forward(#(#inputs,)*);
+            }
+        } else {
+            quote! {
+                let (#(#outputs,)*) = self.#name.forward(#(#inputs,)*);
+            }
+        }
+    }
+
+    /// Generates source code for the forward method results
+    fn forward_method_results(&self) -> TokenStream {
+        let mut outputs = vec![];
+        for output in self.graph.outputs.iter() {
+            let name = Ident::new(&output.name, Span::call_site());
+            outputs.push(quote! {
+                #name
+            });
+        }
+        if outputs.len() == 1 {
+            let output = outputs.pop().unwrap();
+            quote! {
+                #output
+            }
+        } else {
+            quote! {
+                (#(#outputs,)*)
+            }
+        }
+    }
+
+    /// Generates source code for the stateless node calls, i.e. add, mul, etc.
+    fn node_call_stateless(node: &Node, imports: &mut HashSet<String>) -> TokenStream {
+        if node.is_stateful {
+            panic!("Node must be stateless");
+        }
+
+        // let name = Ident::new(&node.name, Span::call_site());
+
+        let mut inputs = vec![];
+
+        for input in node.inputs.iter() {
+            let name = Ident::new(&input.name, Span::call_site());
+            inputs.push(quote! {
+                #name
+            });
+        }
+
+        let mut outputs = vec![];
+
+        for output in node.outputs.iter() {
+            let name = Ident::new(&output.name, Span::call_site());
+            outputs.push(quote! {
+                #name
+            });
+        }
+
+        // let input = inputs.pop().unwrap();
```

Comment:
deep code

Reply:
oh, you mean dead code. OK. I'll remove it.